### DISCUSSION TITLE
Using Skia to create a custom control does not trigger pointer events.

### DISCUSSION THREAD

**Discussion by mccp-dssc** [AvaloniaUI/Avalonia#16186] [Q&A]
I'm starting a new project and would like to use AvaloniaUI. I tried running the RenderDemo from avalonia samples and looked at [CustomSkiaPage.cs](https://github.com/AvaloniaUI/Avalonia/blob/master/samples/RenderDemo/Pages/CustomSkiaPage.cs) to learn how to create custom controls and use Skia to render the control.

However, I'm having difficulty understanding why pointer events are not triggering when I use Skia to render my control.
For example, when copying the code from CustomSkiaPage:
```
public override void Render(DrawingContext context)
{
    context.Custom(new CustomDrawOp(new Rect(0, 0, Bounds.Width, Bounds.Height), _noSkia));
    Dispatcher.UIThread.InvokeAsync(InvalidateVisual, DispatcherPriority.Background);
}
```

And in that same control, I override the OnPointerPressed, i.e.:
```
protected override void OnPointerPressed(PointerPressedEventArgs e)
{
    base.OnPointerPressed(e);
}
```

When I put a breakpoint in that line, that method is not triggered when clicking the control.

Can any body explain what is going on here because I can't find any tutorials or resources to fully explain this.

**Comment by stevemonaco** [comment]
~~Custom drawing itself has no effect on hit testing, AFAIK. If you have a complex shape to hit test, then you should implement `ICustomHitTest`. Otherwise, if it's simply the entire bounds of your control, then ensure there's a `Background`. If it's `null`, then there's nothing to hit test so set it to `Transparent` instead. This is well-known behavior for those familiar with WPF, but I'm not sure if/where it's covered in docs.~~

**Comment by maxkatz6** [comment]
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Rendering/SceneGraph/CustomDrawOperation.cs#L24
